contributingGuides/STYLE.md

```diff
@@ -477,20 +477,63 @@ if (ref.current && 'getBoundingClientRect' in ref.current) {
 
 ### Default value for inexistent IDs
 
- Use `'-1'` or `-1` when there is a possibility that the ID property of an Onyx value could be `null` or `undefined`.
+Use `CONST.DEFAULT_NUMBER_ID` when there is a possibility that the number ID property of an Onyx value could be `null` or `undefined`. **Do not default string IDs to any value unless absolutely necessary**, in case it's necessary use `CONST.DEFAULT_STRING_ID` instead.
 
 ``` ts
 // BAD
-const foo = report?.reportID ?? '';
-const bar = report?.reportID ?? '0';
+const accountID = report?.ownerAccountID ?? -1;
+const accountID = report?.ownerAccountID ?? 0;
+const reportID = report?.reportID ?? '-1';
 
-report ? report.reportID : '0';
-report ? report.reportID : '';
+// BAD
+report ? report.ownerAccountID : -1;
+
+// GOOD
+const accountID = report?.ownerAccountID ?? CONST.DEFAULT_NUMBER_ID;
+const accountID = report?.ownerAccountID ?? CONST.DEFAULT_NUMBER_ID;
+const reportID = report?.reportID;
 
 // GOOD
-const foo = report?.reportID ?? '-1';
+report ? report.ownerAccountID : CONST.DEFAULT_NUMBER_ID;
+```
+
+Here are some common cases you may face when fixing your code to remove the default values.
+
+#### **Case 1**: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
+
+```diff
+-Report.getNewerActions(newestActionCurrentReport?.reportID ?? '-1', newestActionCurrentReport?.reportActionID ?? '-1');
++Report.getNewerActions(newestActionCurrentReport?.reportID, newestActionCurrentReport?.reportActionID);
+```
+
+> error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string'. Type 'undefined' is not assignable to type 'string'.
+
+We need to change `Report.getNewerActions()` arguments to allow `undefined`. By doing that we could add a condition that return early if one of the parameters are falsy, preventing the code (which is expecting defined IDs) from executing.
```

Comment:
Do we? I thought we would pass `CONST.DEFAULT_NUMBER_ID` and thus not allow undefined? (the rest of the code change looks good)

Reply:
This is a string, our guidelines are to allow these to be undefined which is falsey and prevents us from accidentally accessing Onyx values.